File: app/src/main/java/com/lightsoft/parceler/model/Device.java
package com.lightsoft.parceler.model;

import com.lightsoft.parceler.converter.DeviceConverter;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

@Parcel(value = Parcel.Serialization.BEAN, converter = DeviceConverter.class)
public class Device {
    private String manufacturer;
    private String model;
    private String deviceId;

    @ParcelConstructor
    public Device(String manufacturer, String model, String deviceId) {
        this.manufacturer = manufacturer;
        this.model = model;
        this.deviceId = deviceId;
    }

    public Device() {
    }

    public String getManufacturer() {
        return manufacturer;
    }

    public void setManufacturer(String manufacturer) {
        this.manufacturer = manufacturer;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public String getDeviceId() {
        return deviceId;
    }

    public void setDeviceId(String deviceId) {
        this.deviceId = deviceId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;

        Device device = (Device) o;

        return (manufacturer != null ? manufacturer
                .equals(device.manufacturer) : device.manufacturer == null) && (model != null ? model
                .equals(device.model) : device.model == null) && (deviceId != null ? deviceId
                .equals(device.deviceId) : device.deviceId == null);
    }

    @Override
    public int hashCode() {
        int result = manufacturer != null ? manufacturer.hashCode() : 0;
        result = 31 * result + (model != null ? model.hashCode() : 0);
        result = 31 * result + (deviceId != null ? deviceId.hashCode() : 0);
        return result;
    }
}

File: kotlinparceler/src/main/java/com/lightsoft/kotlinparceler/model/Device.kt
package com.lightsoft.kotlinparceler.model

import com.lightsoft.kotlinparceler.converter.DeviceConverter
import org.parceler.Parcel
import org.parceler.ParcelConstructor
import org.parceler.ParcelProperty

@Parcel(Parcel.Serialization.BEAN, converter = DeviceConverter::class)
open class Device
@ParcelConstructor
constructor(@ParcelProperty("manufacturer") var manufacturer: String?,
            @ParcelProperty("model") var model: String?,
            @ParcelProperty("deviceId") var deviceId: String?)
{
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other?.javaClass != javaClass) return false

        other as Device

        if (manufacturer != other.manufacturer) return false
        if (model != other.model) return false
        if (deviceId != other.deviceId) return false

        return true
    }

    override fun hashCode(): Int {
        var result = manufacturer?.hashCode() ?: 0
        result = 31 * result + (model?.hashCode() ?: 0)
        result = 31 * result + (deviceId?.hashCode() ?: 0)
        return result
    }
}